Mathlib/Data/Set/Card.lean

```diff
@@ -1060,5 +1061,45 @@ theorem ncard_eq_three : s.ncard = 3 ↔ ∃ x y z, x ≠ y ∧ x ≠ z ∧ y 
   · rwa [ENat.coe_toNat] at h; rintro h'; simp [h'] at h
   simp [h]
 
+open Cardinal
+
+theorem exists_union_disjoint_cardinal_eq_of_infinite (h : s.Infinite) : ∃ (t u : Set α),
+    t ∪ u = s ∧ Disjoint t u ∧ #t = #u := by
+  have := h.to_subtype
+  have f : s ≃ s ⊕ s := by
+    apply Classical.choice
```

Comment:
Starting the proof with `apply Classical.choice` seems weird. You can prove that the thing exists and then obtain it (since you are proving something Lean should not complain).

Reply:
Fixed!